docs/tutorials/automatic-encryption.txt

```diff
@@ -0,0 +1,211 @@
+
+  For more information about creating a master key, see the
+  `Create a Master Key <https://www.mongodb.com/docs/manual/core/csfle/#a.-create-a-master-key>`_
+  section of the MongoDB manual.
+
+Configure Clients
+=================
+
+Automatic CSFLE requires some additional configuration for the MongoDB client.
+Assuming that your application has just one ``default`` client, you need to
+add the following to your ``mongoid.yml``:
+
+.. code-block:: yaml
+
+  development:
+    clients:
+      default:
+        uri: mongodb+srv://user:pass@yourcluster.mongodb.net/blog_development?retryWrites=true&w=majority
+        options:
+          auto_encryption_options: # This key enables automatic encryption
+            key_vault_namespace: 'encryption.__keyVault' # Database and collection to store data keys
+            kms_providers: # Tells the driver where to obtain master keys
+              local: # We use the local key in this tutorial
+                key: "<%= ENV['CUSTOMER_MASTER_KEY'] %>" # Key that we generated earlier
+            extra_options:
+                crypt_shared_lib_path: '/path/to/mongo_crypt_v1.so' # Only if you use the library
+
+
+Create a Data Encryption Key
+============================
+
+A Data Encryption Key (DEK) is the key you use to encrypt the fields in your
+MongoDB documents. You store your Data Encryption Key in your Key Vault
+collection encrypted with your CMK.
+
+To create a DEK in Mongoid you can use ``db:mongoid:encryption:create_data_key``
+Rake task:
+
+.. code-block:: sh
+
+  % db:mongoid:encryption:create_data_key
+  Created data key with id: 'KImyywsTQWi1+cFYIHdtlA==' for kms provider: 'local' in key vault: 'encryption.__keyVault'.
+
+You can create multiple DEKs, if necessary.
+
+.. code-block:: sh
+
+  % db:mongoid:encryption:create_data_key
+  Created data key with id: 'Vxr5m+5cQISjDOruzZgE0w==' for kms provider: 'local' in key vault: 'encryption.__keyVault'.
+
+
+Configure Encryption Schema
+===========================
+
+Now we can tell Mongoid what should be encrypted:
+
+.. code-block:: ruby
+
+  class Patient
+    include Mongoid::Document
+
+    # Tells Mongoid what DEK should be used to encrypt fields of the document
+    # and its embedded documents.
+    encrypt_with key_id: 'KImyywsTQWi1+cFYIHdtlA=='
+
+    # This field is not encrypted.
+    field :category, type: String
+
+    # This field is encrypted using AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic
+    # algorithm.
+    field :policy_number, type: Integer, encrypt: {
+        deterministic: true
+    }
+    # This field is encrypted using AEAD_AES_256_CBC_HMAC_SHA_512-Random
+    # algorithm.
+    field :blood_type, type: String, encrypt: {
+        deterministic: false
+    }
+    # This field is encrypted using AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic
+    # algorithm and using  a different data key.
+    field :ssn, type: Integer, encrypt: {
+        deterministic: true, data_key_id: 'Vxr5m+5cQISjDOruzZgE0w=='
```

Comment:
```
encrypt_with key_id: ___

vs

field :ssn, type: Integer, encrypt: { deterministic: true, data_key_id: ___
```

It would be good to use either `key_id` or `data_key_id` consistently.

Reply:
Wow, I am wondering how I could missed that. Thank you!